qubesagent/firewall.py

```diff
@@ -62,6 +62,17 @@ def apply_rules(self, source_addr, rules):
         '''Apply rules in given source address'''
         raise NotImplementedError
 
+    def run_firewall_dir(self):
+        '''Run scripts dir contents, before user script'''
+        script_dir_path = '/rw/config/qubes-firewall.d'
```

Comment:
Does it make sense to look _also_ for `/etc/qubes/qubes-firewall.d`? For plugging scripts shipped by some package in template. OTOH, such packages can ship own systemd units and set appropriate ordering (like `Before=network.target` and `After=qubes-firewall.service`)

Reply:
Sure!